staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -199,7 +193,18 @@ func (s *Scheme) AddKnownTypeWithName(gvk schema.GroupVersionKind, obj Object) {
 		panic("All types must be pointers to structs.")
 	}
 
+	if oldT, found := s.gvkToType[gvk]; found && oldT != t {
```

Comment:
> Preserving stack traces is only for debugging, won't be merged.

ah, ok

> Am not so sure about panicking on duplicated registration. We might have situation where it's tricky to avoid repeated registration

tolerating multiple identical registrations to make it idempotent seems fine